### PR TITLE
well-defined parameter types

### PR DESCRIPTION
**Pull request by gaoning777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/978)

### CODE REVIEW

File: sdk/python/kfp/dsl/_types.py [hunk 1]
```diff
@@ -19,72 +19,63 @@ class BaseType:
 
 # Primitive Types
 class Integer(BaseType):
-	openapi_schema_validator = {
+	openAPIV3Schema = {
```
Comment (hongye-sun) [inline sdk/python/kfp/dsl/_types.py]: Could you give me some context on this change?

File: sdk/python/tests/dsl/component_tests.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def _set_metadata(self, component_meta):
         self._metadata = component_meta
 
     @component
-    def componentA(a: {'Schema': {'file_type': 'csv'}}, b: Integer() = 12, c: GCSPath(path_type='file', file_type='tsv') = 'gs://hello/world') -> {'model': Integer()}:
+    def componentA(a: {'Schema': {'file_type': 'csv'}}, b: Integer() = 12, c: {'GCSPath': {'path_type': 'file', 'file_type':'tsv'}} = 'gs://hello/world') -> {'model': Integer()}:
```
Comment (hongye-sun) [inline sdk/python/tests/dsl/component_tests.py]: I thought that you are going to remove this properties for artifact.

File: sdk/python/kfp/dsl/_types.py (hunk 1 above)
Comment (gaoning777) [reply sdk/python/kfp/dsl/_types.py]: I changed the key name to openAPIV3Schema for two reasons:
1) consistent with the static-config design of Pascal.
2) as a conventional name, openAPIV3Schema looks more official. 
WDYT?

File: sdk/python/tests/dsl/component_tests.py (hunk 2 above)
Comment (gaoning777) [reply sdk/python/tests/dsl/component_tests.py]: Yes, from all the user facing interfaces. 
I kept the unit tests to verify if the codebase works with the property lists.
I'm ok with removing them if it causes confusion, though.

Comment (hongye-sun) [reply sdk/python/tests/dsl/component_tests.py]: Maybe rename the properties to be not artifact related.

File: sdk/python/kfp/dsl/_types.py (hunk 1 above)
Comment (hongye-sun) [reply sdk/python/kfp/dsl/_types.py]: I feel it's better to follow python naming convention and only use the camel case name when serializing it to yaml or json.

File: sdk/python/tests/dsl/component_tests.py (hunk 2 above)
Comment (gaoning777) [reply sdk/python/tests/dsl/component_tests.py]: OK, let me do that.

Comment (gaoning777) [reply sdk/python/tests/dsl/component_tests.py]: Done.

File: sdk/python/kfp/dsl/_types.py (hunk 1 above)
Comment (hongye-sun) [reply sdk/python/kfp/dsl/_types.py]: This comment is actually nit. I will let you decide.

/lgtm

Comment (gaoning777) [reply sdk/python/kfp/dsl/_types.py]: Maybe revert to the lower case with underscores, as indicated in the google python style guide(http://google.github.io/styleguide/pyguide.html)?

Comment (gaoning777) [reply sdk/python/kfp/dsl/_types.py]: And, having the yaml format the same lower case with underscore?

Comment (gaoning777) [reply sdk/python/kfp/dsl/_types.py]: Done.

**Pull request merged**